libs/promises/api/celix/impl/SharedPromiseState.h

```diff
@@ -106,9 +107,11 @@ namespace celix::impl {
 
         std::shared_ptr<celix::IExecutor> getExecutor() const;
 
+        std::shared_ptr<celix::IScheduledExecutor> getScheduledExecutor() const;
```

Comment:
This function (and coincidentally the `getExecutor` function that was already there, are good examples of being marked with `[[nodiscard]]`

Reply:
added